server/src/internalClusterTest/java/org/opensearch/search/nested/SimpleNestedIT.java

```diff
@@ -497,7 +497,13 @@ public void testExplain() throws Exception {
         assertThat(searchResponse.getHits().getTotalHits().value, equalTo(1L));
         Explanation explanation = searchResponse.getHits().getHits()[0].getExplanation();
         assertThat(explanation.getValue(), equalTo(searchResponse.getHits().getHits()[0].getScore()));
-        assertThat(explanation.toString(), startsWith("0.36464313 = Score based on 2 child docs in range from 0 to 1"));
+        // Concurrent search will have deleted doc count, hence the assertion string will be different
+        // ref: https://github.com/opensearch-project/OpenSearch/issues/11413
+        if (isConcurrentSearchEnabled()) {
+            assertThat(explanation.toString(), startsWith("0.36464313 = Score based on 2 child docs in range"));
+        } else {
+            assertThat(explanation.toString(), startsWith("0.36464313 = Score based on 2 child docs in range from 0 to 1"));
+        }
```

Comment:
Do we really need to check the range in the non-concurrent case? And if not we also don't need the `isConcurrentSearchEnabled` logic then?

If we do need to check the range then shouldn't we do it for the concurrent search case as well? Not sure if it's possible to calculate it based on the number of randomly ingested + deleted docs.

Reply:
Lets have 2 separate test cases: one with deleted docs and other without deleted docs. Then in both the tests the explain output should probably be same and assertions will be same for both concurrent and non-concurrent path.

Reply:
updated the PR by introducing three different test cases: 
- single doc indexed with no deleted docs by removing indexRandomForConcuurentSearch (assertion for explain API remains same in this case)
- multiple docs indexed with no deleted docs (assertion for explain API changes in this case)
- single doc indexed with deleted docs by adding indexRandomForConcuurentSearch (assertion for explain API changes in this case)